esphome/components/atm90e32/number/__init__.py

```diff
@@ -0,0 +1,137 @@
+import esphome.codegen as cg
+from esphome.components import number
+import esphome.config_validation as cv
+from esphome.const import (
+    CONF_ID,
+    CONF_NAME,
+    CONF_PHASE_A,
+    CONF_PHASE_B,
+    CONF_PHASE_C,
+    CONF_REFERENCE_VOLTAGE,
+    ENTITY_CATEGORY_CONFIG,
+    UNIT_AMPERE,
+    UNIT_VOLT,
+)
+
+from .. import atm90e32_ns
+from ..sensor import ATM90E32Component
+
+ATM90E32Number = atm90e32_ns.class_(
+    "ATM90E32Number", number.Number, cg.Parented.template(ATM90E32Component)
+)
+
+CONF_REFERENCE_CURRENT = "reference_current"
+PHASE_KEYS = [CONF_PHASE_A, CONF_PHASE_B, CONF_PHASE_C]
+
+
+def reference_voltage_schema():
+    return cv.All(
+        cv.Schema(
+            {
+                cv.Required(CONF_NAME): cv.string,
+                cv.Optional("mode", default="box"): cv.string,
+                cv.Optional("optimistic", default=True): cv.boolean,
+                cv.Optional("min_value", default=100.0): cv.float_,
+                cv.Optional("max_value", default=260.0): cv.float_,
+                cv.Optional("step", default=0.1): cv.float_,
+            }
+        ).extend(
+            number.number_schema(
+                class_=ATM90E32Number,
+                unit_of_measurement=UNIT_VOLT,
+                entity_category=ENTITY_CATEGORY_CONFIG,
+                icon="mdi:power-plug",
+            )
+        )
+    )
+
+
+def reference_current_schema():
+    return cv.All(
```

Comment:
This doesnt need to be a method, it can be static
```suggestion
REFERENCE_CURRENT_SCHEMA = cv.All(
```